### PR TITLE
850 feauture add calendar display options

### LINKED ISSUE #850: Add calendar display options
Now we have only one display option - the whole week.

image

You need to add the ability to select a display option. As in the example below: 
(Working week and week)

image
image
image

You also need to change the width of the columns in the calendar for Saturday and Sunday. Now these columns are smaller than others and the events in them are displayed in a compressed form. Make all columns the same width.

image

### PR DESCRIPTION
**Pull request by BobTheSuperman**
added the ability to chose a calendar display option (Working week or full week)

### CODE REVIEW

File: CharlieBackend.Panel/HtmlHelpers/CalendarHelper.cs [hunk 1]
```diff
@@ -11,6 +11,24 @@ namespace CharlieBackend.Panel.HtmlHelpers
...
+
+            foreach (DayOfWeek day in Enum.GetValues(typeof(DayOfWeek)))
+            {
+                TagBuilder div = GetDayBlock(day, displayType);
+
+                div.InnerHtml.Append(day.ToString());
+
+                using var writer = new System.IO.StringWriter();
+                div.WriteTo(writer, HtmlEncoder.Default);
+                result += writer.ToString();
```
Comment (PavelTsyposhchuk) [inline CharlieBackend.Panel/HtmlHelpers/CalendarHelper.cs]: suggest use stringbuilder instead string, of course it`s not a very long loop but anyway

File: CharlieBackend.Panel/HtmlHelpers/CalendarHelper.cs [hunk 2]
```diff
@@ -190,6 +221,27 @@ private static TagBuilder GetButtonContainerHtml(CalendarScheduledEventModel mod
...
+            string dayOfWeekClass = string.Empty;
+
+            switch (displayType)
+            {
+                case CalendarDisplayType.FullWeek:
+                    dayOfWeekClass = IsWeekend(dayOfWeek) ? "col-calendar-full-week weekend" : "col-calendar-full-week";
+                    break;
+
+                case CalendarDisplayType.WorkingWeek:
+                    dayOfWeekClass = IsWeekend(dayOfWeek) ? "d-none weekend" : "col-calendar-working-week";
```
Comment (PavelTsyposhchuk) [inline CharlieBackend.Panel/HtmlHelpers/CalendarHelper.cs]: What about extracting text to private constants?

File: CharlieBackend.Panel/HtmlHelpers/CalendarHelper.cs [hunk 3]
```diff
@@ -190,6 +221,27 @@ private static TagBuilder GetButtonContainerHtml(CalendarScheduledEventModel mod
...
+                    break;
+
+                default:
+                    goto case CalendarDisplayType.WorkingWeek;
```
Comment (PavelTsyposhchuk) [inline CharlieBackend.Panel/HtmlHelpers/CalendarHelper.cs]: goto is very bad practice, better do a method something like GetDayOfFullWeekClass with this code `dayOfWeekClass = IsWeekend(dayOfWeek) ? "col-calendar-full-week weekend" : "col-calendar-full-week";` and call it there

File: CharlieBackend.Panel/Views/Calendar/Index.cshtml [hunk 4]
```diff
@@ -70,6 +89,21 @@
...
+            if (x == 1) {
+                $(".weekend").removeClass("col-calendar-full-week").addClass("d-none");
+                $(".col-calendar-full-week").removeClass("col-calendar-full-week").addClass("col-calendar-working-week");
+            }
+            else if (x == 2) {
+                $(".col-calendar-working-week").removeClass("col-calendar-working-week").addClass("col-calendar-full-week");
+                $(".weekend").removeClass("d-none").addClass("col-calendar-full-week");
+            }
+        }
+    </script>
```
Comment (PavelTsyposhchuk) [inline CharlieBackend.Panel/Views/Calendar/Index.cshtml]: same as css, I don't think that we should have js code there, we have this file for example to store js script and then just include it from ~\WhatBackend\CharlieBackend.Panel\wwwroot\js\sb-admin-2.js

File: CharlieBackend.Panel/Views/Calendar/Index.cshtml [hunk 5]
```diff
@@ -34,11 +35,29 @@
...
+
+    .col-1:first-child, .col-2, .col-calendar-full-week, .col-calendar-working-week {
         border-right: 0.5px solid;
     }
 
-    .col-2, .col-1 {
+    .col-2, .col-1, .col-calendar-full-week, .col-calendar-working-week {
         border-bottom: 0.5px solid grey;
         min-height: 100px;
     }
```
Comment (PavelTsyposhchuk) [inline CharlieBackend.Panel/Views/Calendar/Index.cshtml]: IMHO this should be in file ~\WhatBackend\CharlieBackend.Panel\wwwroot\css\cssEdiary.css or site.css in same folder

**Review by PavelTsyposhchuk — CHANGES_REQUESTED** [review]

File: CharlieBackend.Panel/HtmlHelpers/CalendarHelper.cs (hunk 1 above)
Comment (BobTheSuperman) [reply CharlieBackend.Panel/HtmlHelpers/CalendarHelper.cs]: Done!

File: CharlieBackend.Panel/HtmlHelpers/CalendarHelper.cs (hunk 2 above)
Comment (BobTheSuperman) [reply CharlieBackend.Panel/HtmlHelpers/CalendarHelper.cs]: I agree with you! Done.

File: CharlieBackend.Panel/HtmlHelpers/CalendarHelper.cs (hunk 3 above)
Comment (BobTheSuperman) [reply CharlieBackend.Panel/HtmlHelpers/CalendarHelper.cs]: Done

File: CharlieBackend.Panel/wwwroot/js/calendar.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
+﻿var seGroup = null;
+var seMentor = null;
+var seTheme = null;
+var seTime = null;
+var seEventOccurrenceId = null;
+var seSingleEventId = null;
```
Comment (PavelTsyposhchuk) [inline CharlieBackend.Panel/wwwroot/js/calendar.js]: creating var variable outside of functions makes its global scope visible, I don't think that it was ur purpose

**Review by PavelTsyposhchuk — CHANGES_REQUESTED** [review]

**Review by BohdanZaliubovskyi — APPROVED** [review]

**Review by PavelTsyposhchuk — APPROVED** [review]

**Review by andy-dendy-repo — APPROVED** [review]

**Review by Ssashaa — APPROVED** [review]

**Review by DevPew01 — APPROVED** [review]

**Pull request merged**